cpp/src/parquet/column_reader.cc

```diff
@@ -50,6 +51,141 @@ using arrow::internal::checked_cast;
+        valid_bits_writer.Clear();
+        *null_count += 1;
+      } else {
+        throw ParquetException("definition level exceeds maximum");
+      }
+    }
+
+    valid_bits_writer.Next();
+  }
+  valid_bits_writer.Finish();
+  *values_read = valid_bits_writer.position();
+}
+#endif
+
+template <bool has_repeated_parent>
+void DefinitionLevelsToBitmapSimd(const int16_t* def_levels, int64_t num_def_levels,
+                                  const int16_t required_definition_level,
+                                  int64_t* values_read, int64_t* null_count,
+                                  uint8_t* valid_bits, int64_t valid_bits_offset) {
+  constexpr int64_t kBitMaskSize = 64;
+  int64_t set_count = 0;
+  *values_read = 0;
+  while (num_def_levels > 0) {
+    int64_t batch_size = std::min(num_def_levels, kBitMaskSize);
+    CheckLevelRange(def_levels, batch_size, required_definition_level);
+    uint64_t defined_bitmap = internal::GreaterThanBitmap(def_levels, batch_size,
+                                                          required_definition_level - 1);
+    if (has_repeated_parent) {
+      // This is currently a specialized code path assuming only (nested) lists
+      // present through the leaf (i.e. no structs).
+      // Upper level code only calls this method
+      // when the leaf-values are nullable (otherwise no spacing is needed),
+      // Because only nested lists exists it is sufficient to know that the field
+      // was either null or included it (i.e. >= previous definition level -> > previous
+      // definition - 1). If there where structs mixed in, we need to know the def_level
+      // of the repeated parent so we can check for def_level > "def level of repeated
+      // parent".
+      uint64_t present_bitmap = internal::GreaterThanBitmap(
+          def_levels, batch_size, required_definition_level - 2);
+      *values_read += internal::AppendSelectedBitsToValidityBitmap(
+          /*new_bits=*/defined_bitmap,
+          /*selection_bitmap*/ present_bitmap, valid_bits, &valid_bits_offset,
+          &set_count);
+    } else {
+      internal::AppendToValidityBitmap(
+          /*new_bits=*/defined_bitmap,
+          /*new_bit_count=*/batch_size, valid_bits, &valid_bits_offset, &set_count);
+      *values_read += batch_size;
+    }
+    def_levels += batch_size;
+    num_def_levels -= batch_size;
+  }
+  *null_count += *values_read - set_count;
+}
+
+inline void DefinitionLevelsToBitmapDispatch(
+    const int16_t* def_levels, int64_t num_def_levels, const int16_t max_definition_level,
+    const int16_t max_repetition_level, int64_t* values_read, int64_t* null_count,
+    uint8_t* valid_bits, int64_t valid_bits_offset) {
+  if (max_repetition_level > 0) {
+#if ARROW_LITTLE_ENDIAN
+// we use AVx2 as a proxy for BMI2 instruction set, since there doesn't seem to be a clean
+// way o detect that latter for MSVC.
+#if defined(ARROW_HAVE_BMI2)
+    // We need BIM2 instruction which is AVX2 should imply.
```

Comment:
nit: AVx2 -> AVX2, o detect -> to detect, BIM2 -> BMI2

Reply:
Remove the comment as it was stale.